### PR TITLE
python3Packages.pmdarima: init at 2.0.3

### PR DESCRIPTION
**Pull request by mbalatsko**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Results of `nixpkgs-review pr 255700` on x86_64-linux:
```
2 packages built:
python310Packages.pmdarima python311Packages.pmdarima
```

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.pmdarima python311Packages.pmdarima

**Comment by mbalatsko** [comment]
Dependency of https://github.com/NixOS/nixpkgs/issues/247089

File: pkgs/development/python-modules/pmdarima/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  # package, so that cython's compiled files are available.
+  preCheck = ''
+    cd $out/lib/${python.libPrefix}/site-packages
+  '';
+
+  nativeCheckInputs = [ pytestCheckHook ];
+  pytestFlagsArray = [
+    # toches internet
+    "--deselect=pmdarima/datasets/tests/test_load_datasets.py::test_load_from_web"
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pmdarima/default.nix:53]: `disabledTests` and `disabledTestPaths` are helpers to disable tests.

```suggestion
  disabledTests= [
    # touches internet
    "test_load_from_web"
  ];
```

File: pkgs/development/python-modules/pmdarima/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  pythonImportsCheck = [ "pmdarima" ];
+
+  meta = with lib; {
+    description = "A statistical library designed to fill the void in Python's time series analysis capabilities, including the equivalent of R's auto.arima function";
+    homepage = "https://github.com/alkaline-ml/pmdarima";
```
Comment (fabaff) [inline pkgs/development/python-modules/pmdarima/default.nix:59]: Tools (e.g., `nix-update`) can pick-up the changelog entry and add it to the commit message.

```suggestion
    homepage = "https://github.com/alkaline-ml/pmdarima";
    changelog = "https://github.com/alkaline-ml/pmdarima/releases/tag/v${version}";
```

Comment (mbalatsko) [reply pkgs/development/python-modules/pmdarima/default.nix:59]: Interesting, I used `nix-init` to generate the first version of `default.nix` and it didn't produce the changelog attr here 🤔

Comment (mbalatsko) [reply pkgs/development/python-modules/pmdarima/default.nix:59]: Fixed

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 255700` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.pmdarima</li>
    <li>python310Packages.pmdarima.dist</li>
    <li>python311Packages.pmdarima</li>
    <li>python311Packages.pmdarima.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Pull request merged**